### PR TITLE
Use (void) instead of () in function signatures

### PR DESCRIPTION
**Pull request by trufae**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

It's incorrect to define functions with no parameters like (), it is necessary to use (void) instead

This PR updates SDB ( i fixed that in sdb too), and the bug is also happening in tree-sitter, so @ret2libc may push the fix in the official repo or my fix will break on the next sync.

**Test plan**

```
git grep -e ^R_API -e ^static | grep '() {'
```

**Closing issues**

None, just 428 warnings less

### CODE REVIEW

**Comment by trufae** [comment]
Beautiful

Screenshot 2020-06-09 at 20 19 53

File: shlr/tree-sitter/lib/src/query.c [hunk 1]
```diff
@@ -259,7 +259,7 @@ static void stream_scan_identifier(Stream *stream) {
  * CaptureListPool
  ******************/
 
-static CaptureListPool capture_list_pool_new() {
+static CaptureListPool capture_list_pool_new(void) {
```
Comment (ret2libc) [inline shlr/tree-sitter/lib/src/query.c:262]: It's better if we don't touch these files but just work upstream. I will work with tree-sitter to include these fixes.

**Comment by radare** [comment]
I know. I warned about this when i filled the issue. I touched them because the test will fail otherwise. Can you submit this fix in treesitter?

**Comment by ret2libc** [comment]
> I know. I warned about this when i filled the issue. I touched them because the test will fail otherwise. Can you submit this fix in treesitter?

Apparently they already fixed it, 7 days ago. https://github.com/tree-sitter/tree-sitter/commit/81d533d2d1b580fdb507accabc91ceddffb5b6f0  . They haven't done a release yet though.

**Comment by XVilka** [comment]
Can you just ignore that directory in the test? If not, and the required changes in the tree-sitter already, I think it's fine to merge as is, until the tree-sitter will make a new release.

**Comment by ret2libc** [comment]
If we can't wait for a release, update the `TS_TIP` in shlr/makefile and shlr/meson.build and run a `make tree-sitter-sync`. It's better than applying random fixes.

**Comment by trufae** [comment]
i've reverted the treesitter fixes and the test is only checking in libr/

**Comment by trufae** [comment]
anything left?

**Pull request merged**